src/main/java/by/yahor_kulesh/dao/TicketDAO.java

```diff
@@ -1,122 +1,63 @@
 package by.yahor_kulesh.dao;
 
-import by.yahor_kulesh.config.ConnectionConfig;
-import by.yahor_kulesh.model.tickets.BusTicket;
-import by.yahor_kulesh.model.tickets.ConcertTicket;
-import by.yahor_kulesh.model.tickets.Ticket;
-import by.yahor_kulesh.utils.ObjectArray;
+import by.yahor_kulesh.config.SessionFactoryProvider;
+import by.yahor_kulesh.entity.TicketEntity;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.SelectionQuery;
 
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Timestamp;
-import java.time.ZoneId;
+import java.util.List;
 import java.util.UUID;
 
 public class TicketDAO{
-    private ResultSet rs = null;
 
-    public void insert(Ticket ticket){
-        try(
-                Connection con = ConnectionConfig.getConnection();
-                PreparedStatement stat = con.prepareStatement("INSERT INTO ticket(id,user_id,ticket_type,creation_date) VALUES (?,?,CAST(? AS ticket_type),?)")
-        ){
-            prepareTicketForStatement(ticket, stat);
-            stat.execute();
-        }catch(SQLException e){
-            System.err.println(e.getMessage());
+    public void saveTicket(TicketEntity ticket){
+        try(Session session = SessionFactoryProvider.getSessionFactory().openSession()){
+            Transaction tx = session.beginTransaction();
```

Comment:
do not cut variables names: 
`Transaction transaction `
instead of 
`Transaction tx`